WireDomain/Sources/WireDomain/Repositories/UserClients/UserClientsRepository.swift

```diff
@@ -112,22 +129,20 @@ public struct ClientRepository: ClientRepositoryProtocol {
             }
         }
 
-        try context.save()
-
         return localUserClient
     }
 
     public func updateClient(
         with id: String,
-        from remoteClient: WireAPI.UserClient,
+        from remoteClient: WireAPI.SelfUserClient,
         isNewClient: Bool
     ) async throws {
         guard let localClient = UserClient.fetchExistingUserClient(
             with: id,
             in: context
         ) else {
             return WireLogger.userClient.error(
-                "Failed to find existing client with id: \(id)"
+                "Failed to find existing client with id: \(id.obfuscated())"
```

Comment:
note we should always use the same method to obfuscate the data, also when using datadog this id should be stable to be used through debugging. Prefer: `redactedAndTruncated()`

Reply:
fixed in https://github.com/wireapp/wire-ios/pull/2019/commits/f606230d354cb05894ca1ce4dc3fa8c1f3d0f244